src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -922,14 +922,27 @@ def _resolve_policy_id(policy, policy_set_definition, client):
     policy_id = policy or policy_set_definition
     if not is_valid_resource_id(policy_id):
         if policy:
-            policy_def = client.policy_definitions.get(policy)
+            policy_def = _get_custom_or_builtin_policy(client, policy)
             policy_id = policy_def.id
         else:
-            policy_set_def = client.policy_set_definitions.get(policy_set_definition)
+            policy_set_def = _get_custom_or_builtin_policy(client, policy_set_definition, 'setdefinition')
```

Comment:
nit, should be: policy_set_def = _get_custom_or_builtin_policy(client, policy_set_definition, `true`)